### PR TITLE
build: svelte v4

### PR DESCRIPTION
**Pull request by peterpeterparker**
# Motivation

Migrate to Svelte v4.

# Blockers

- `svelte-jester`: https://github.com/svelteness/svelte-jester/issues/121
- `MDsvex`: https://github.com/pngwn/MDsveX/issues/528 (we can workaround this one)

# Changes

- migrate to Svelte v4
- adapt interfaces, lint etc. accordingly

### CODE REVIEW

File: src/routes/(page)/styling/usage/+page.md [hunk 1]
```diff
@@ -5,9 +5,9 @@ Within a `script` tag in your layout or app, import the `global.scss` of the lib
 > The style provided by the kit is not pre-compiled - i.e. is not provided as `CSS` files but as `SCSS`. Make sure that a SASS preprocessing is installed in your project.
 
 ```html
-<style lang="scss" global>
+<_style lang="scss" global>
```
Comment (lmuntaner) [inline src/routes/(page)/styling/usage/+page.md:8]: Why is this needed?

File: src/tests/lib/components/DropdownTest.svelte [hunk 2]
```diff
@@ -10,6 +10,7 @@
   };
 </script>
 
+<!-- eslint-disable svelte/valid-compile -->
```
Comment (lmuntaner) [inline src/tests/lib/components/DropdownTest.svelte:13]: What's this?

File: package.json [hunk 3]
```diff
@@ -81,7 +81,7 @@
   ],
   "peerDependencies": {
     "@dfinity/utils": "*",
-    "svelte": "^3.59.1"
+    "svelte": "^4.2.1"
```
Comment (lmuntaner) [inline package.json:84]: 🎉

File: .eslintrc.cjs [hunk 4]
```diff
@@ -1,26 +1,44 @@
...
+  overrides: [
+    {
+      files: ["*.svelte"],
+      parser: "svelte-eslint-parser",
+      parserOptions: {
+        parser: "@typescript-eslint/parser",
+      },
+    },
+  ],
+  rules: {
```
Comment (lmuntaner) [inline .eslintrc.cjs:31]: Was this needed for the migration?

**Review by lmuntaner — APPROVED** [review]
A few comments, but already apprving!

File: src/routes/(page)/styling/usage/+page.md (hunk 1 above)
Comment (peterpeterparker) [reply src/routes/(page)/styling/usage/+page.md:8]: See PR description

File: .eslintrc.cjs (hunk 4 above)
Comment (peterpeterparker) [reply .eslintrc.cjs:31]: Because eslint throw errors which previously were ignored.

File: src/tests/lib/components/DropdownTest.svelte (hunk 2 above)
Comment (peterpeterparker) [reply src/tests/lib/components/DropdownTest.svelte:13]: Disabling an eslint rule in a test component for a test

**Comment by lmuntaner** [comment]
Isn't the blocker of `svelte-jester` solved? https://github.com/svelteness/svelte-jester/issues/121#issuecomment-1648709963

If it's a blocker, shouldn't we set this to Draft PR?

**Comment by peterpeterparker** [comment]
We are using vitest
